provider/internal/java/util.go

```diff
@@ -13,25 +13,66 @@ import (
 	"github.com/go-logr/logr"
 )
 
-// decompileJava is a function that extracts the contents ofa Java archive (.jar|.war|.ear) and
-// decompiles any .class files found using the fernflower decompiler and returns the path to
-// the destition directory on success, error otherwise.
-func decompileJava(ctx context.Context, log logr.Logger, location string) (string, error) {
-	// Get the permissions of the Java archive file
-	fileInfo, err := os.Stat(location)
+const javaProjectPom = `<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>io.konveyor</groupId>
+  <artifactId>java-project</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <name>java-project</name>
+  <url>http://www.konveyor.io</url>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+
+  <dependencies>
+  </dependencies>
+
+  <build>
+  </build>
+</project>
+`
+
+// decompileJava unpacks archive at archivePath, decompiles all .class files in it
+// creates new java project and puts the java files in the tree of the project
+// returns path to exploded archive, path to java project, and an error when encountered
+func decompileJava(ctx context.Context, log logr.Logger, archivePath string) (explodedPath, projectPath string, err error) {
+	projectPath = filepath.Join(filepath.Dir(archivePath), "java-project")
+
+	err = createJavaProject(ctx, projectPath)
 	if err != nil {
-		return "", err
+		log.Error(err, "failed to create java project", "path", projectPath)
+		return
+	} else {
+		log.V(5).Info("created java project", "path", projectPath)
+	}
+
+	explodedPath, err = decompile(ctx, log, archivePath, projectPath)
+	if err != nil {
+		log.Error(err, "failed to decompile archive", "path", archivePath)
+		return
 	}
 
+	return
+}
+
+// decompile is a function that extracts the contents of a Java archive (.jar|.war|.ear) and
+// decompiles any .class files found using the fernflower decompiler into java project location
+// maintaining the tree. swallows decomp and copy errors, returns others
+func decompile(ctx context.Context, log logr.Logger, archivePath, projectPath string) (string, error) {
 	// Create the destDir directory using the same permissions as the Java archive file
 	// java.jar should become java-jar-decompiled
-	destDir := filepath.Join(path.Dir(location), strings.Replace(path.Base(location), ".", "-", -1) + "-decompiled")
-	err = os.MkdirAll(destDir, fileInfo.Mode())
+	destDir := filepath.Join(path.Dir(archivePath), strings.Replace(path.Base(archivePath), ".", "-", -1)+"-decompiled")
+	err := os.MkdirAll(destDir, 0755)
```

Comment:
This should make it so you only set the execute bit for everyone on directories in the case it's missing:

```suggestion
	err := os.MkdirAll(destDir, fileInfo.Mode()|0111)
```